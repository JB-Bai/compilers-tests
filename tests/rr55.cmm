float xa7UqhmTTwb(int DpLM9_QzMB, struct{
}
qfTwt_h[030275][07040][037626][191], struct vP8umFrB WrylQQi[020]){
	int T0mnrAot[73343];
	735.3802e-0[zFtNPqEG3];
	{
		int gNFN48_UOT = EviLQNtWT, dsdtTxrmQ;
		float gIIo1kQSE = 2894, hg9uL1Cj[3585890] = 9203.2609.WyND, cFtIFNCbm[0x3bf];
		struct LhSm1{
		}
		GJhgAtK2, _bRb8p0OM[0x42B][484837][0xd9Dc], m56iXCBaZ = - 451570, kscUtRKIVv, OoIywKD;
		if(rY_KoUTCE(INTFjGf, ! 80.467) || 61.2956e1) if(! AFXuLfIVAu(80.9E-5)[VeckdcHW3(2.94 * CMc8)]) if(704) if(! - fe7mlt(0xdbE,(7886541)) + ytsaD9fl) if(3021) if(02305) 100.91;
		else return irfX6;
		else while(66.4191E0) return 0x1E;
		else if(FFw1o(V4Qq0V, 455)) 55.38;
		yh9YYIYG;
		0xe8f[U4poC7pfI()];
		return QJAWa;
	}
	! - - S2W13d2UQ0N = 16.48 && GZnSUh && 65.56e5[BdFXXIX];
}
int Lr4OhDgMQJ(struct UCB1uF7{
}
Hd0RoYT, struct{
}
Eww7fpa){
	Pnpnaru <= 68.164 -(Oru0yKq) || 0205000 && 607.696e-4;
	0425;
	- 0x3B[952] || Fe7EeDV() +(33.0697E+4) * 89.2 - JwbWpum > 0x520 - - ! ZMJYsgg() > jt9Kb8 || 7.711;
}
float;

