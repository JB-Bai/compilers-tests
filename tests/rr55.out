Program (1)
  ExtDefList (1)
    ExtDef (1)
      Specifier (1)
        TYPE: float
      FunDec (1)
        ID: xa7UqhmTTwb
        LP
        VarList (1)
          ParamDec (1)
            Specifier (1)
              TYPE: int
            VarDec (1)
              ID: DpLM9_QzMB
          COMMA
          VarList (1)
            ParamDec (1)
              Specifier (1)
                StructSpecifier (1)
                  STRUCT
                  LC
                  RC
              VarDec (3)
                VarDec (3)
                  VarDec (3)
                    VarDec (3)
                      VarDec (3)
                        ID: qfTwt_h
                      LB
                      INT: 12477
                      RB
                    LB
                    INT: 3616
                    RB
                  LB
                  INT: 16278
                  RB
                LB
                INT: 191
                RB
            COMMA
            VarList (3)
              ParamDec (3)
                Specifier (3)
                  StructSpecifier (3)
                    STRUCT
                    Tag (3)
                      ID: vP8umFrB
                VarDec (3)
                  VarDec (3)
                    ID: WrylQQi
                  LB
                  INT: 16
                  RB
        RP
      CompSt (3)
        LC
        DefList (4)
          Def (4)
            Specifier (4)
              TYPE: int
            DecList (4)
              Dec (4)
                VarDec (4)
                  VarDec (4)
                    ID: T0mnrAot
                  LB
                  INT: 73343
                  RB
            SEMI
        StmtList (5)
          Stmt (5)
            Exp (5)
              Exp (5)
                FLOAT: 735.380188
              LB
              Exp (5)
                ID: zFtNPqEG3
              RB
            SEMI
          StmtList (6)
            Stmt (6)
              CompSt (6)
                LC
                DefList (7)
                  Def (7)
                    Specifier (7)
                      TYPE: int
                    DecList (7)
                      Dec (7)
                        VarDec (7)
                          ID: gNFN48_UOT
                        ASSIGNOP
                        Exp (7)
                          ID: EviLQNtWT
                      COMMA
                      DecList (7)
                        Dec (7)
                          VarDec (7)
                            ID: dsdtTxrmQ
                    SEMI
                  DefList (8)
                    Def (8)
                      Specifier (8)
                        TYPE: float
                      DecList (8)
                        Dec (8)
                          VarDec (8)
                            ID: gIIo1kQSE
                          ASSIGNOP
                          Exp (8)
                            INT: 2894
                        COMMA
                        DecList (8)
                          Dec (8)
                            VarDec (8)
                              VarDec (8)
                                ID: hg9uL1Cj
                              LB
                              INT: 3585890
                              RB
                            ASSIGNOP
                            Exp (8)
                              Exp (8)
                                FLOAT: 9203.260742
                              DOT
                              ID: WyND
                          COMMA
                          DecList (8)
                            Dec (8)
                              VarDec (8)
                                VarDec (8)
                                  ID: cFtIFNCbm
                                LB
                                INT: 959
                                RB
                      SEMI
                    DefList (9)
                      Def (9)
                        Specifier (9)
                          StructSpecifier (9)
                            STRUCT
                            OptTag (9)
                              ID: LhSm1
                            LC
                            RC
                        DecList (11)
                          Dec (11)
                            VarDec (11)
                              ID: GJhgAtK2
                          COMMA
                          DecList (11)
                            Dec (11)
                              VarDec (11)
                                VarDec (11)
                                  VarDec (11)
                                    VarDec (11)
                                      ID: _bRb8p0OM
                                    LB
                                    INT: 1067
                                    RB
                                  LB
                                  INT: 484837
                                  RB
                                LB
                                INT: 55772
                                RB
                            COMMA
                            DecList (11)
                              Dec (11)
                                VarDec (11)
                                  ID: m56iXCBaZ
                                ASSIGNOP
                                Exp (11)
                                  MINUS
                                  Exp (11)
                                    INT: 451570
                              COMMA
                              DecList (11)
                                Dec (11)
                                  VarDec (11)
                                    ID: kscUtRKIVv
                                COMMA
                                DecList (11)
                                  Dec (11)
                                    VarDec (11)
                                      ID: OoIywKD
                        SEMI
                StmtList (12)
                  Stmt (12)
                    IF
                    LP
                    Exp (12)
                      Exp (12)
                        ID: rY_KoUTCE
                        LP
                        Args (12)
                          Exp (12)
                            ID: INTFjGf
                          COMMA
                          Args (12)
                            Exp (12)
                              NOT
                              Exp (12)
                                FLOAT: 80.467003
                        RP
                      OR
                      Exp (12)
                        FLOAT: 612.955994
                    RP
                    Stmt (12)
                      IF
                      LP
                      Exp (12)
                        NOT
                        Exp (12)
                          Exp (12)
                            ID: AFXuLfIVAu
                            LP
                            Args (12)
                              Exp (12)
                                FLOAT: 0.000809
                            RP
                          LB
                          Exp (12)
                            ID: VeckdcHW3
                            LP
                            Args (12)
                              Exp (12)
                                Exp (12)
                                  FLOAT: 2.940000
                                STAR
                                Exp (12)
                                  ID: CMc8
                            RP
                          RB
                      RP
                      Stmt (12)
                        IF
                        LP
                        Exp (12)
                          INT: 704
                        RP
                        Stmt (12)
                          IF
                          LP
                          Exp (12)
                            Exp (12)
                              NOT
                              Exp (12)
                                MINUS
                                Exp (12)
                                  ID: fe7mlt
                                  LP
                                  Args (12)
                                    Exp (12)
                                      INT: 3518
                                    COMMA
                                    Args (12)
                                      Exp (12)
                                        LP
                                        Exp (12)
                                          INT: 7886541
                                        RP
                                  RP
                            PLUS
                            Exp (12)
                              ID: ytsaD9fl
                          RP
                          Stmt (12)
                            IF
                            LP
                            Exp (12)
                              INT: 3021
                            RP
                            Stmt (12)
                              IF
                              LP
                              Exp (12)
                                INT: 1221
                              RP
                              Stmt (12)
                                Exp (12)
                                  FLOAT: 100.910004
                                SEMI
                              ELSE
                              Stmt (13)
                                RETURN
                                Exp (13)
                                  ID: irfX6
                                SEMI
                            ELSE
                            Stmt (14)
                              WHILE
                              LP
                              Exp (14)
                                FLOAT: 66.419098
                              RP
                              Stmt (14)
                                RETURN
                                Exp (14)
                                  INT: 30
                                SEMI
                          ELSE
                          Stmt (15)
                            IF
                            LP
                            Exp (15)
                              ID: FFw1o
                              LP
                              Args (15)
                                Exp (15)
                                  ID: V4Qq0V
                                COMMA
                                Args (15)
                                  Exp (15)
                                    INT: 455
                              RP
                            RP
                            Stmt (15)
                              Exp (15)
                                FLOAT: 55.380001
                              SEMI
                  StmtList (16)
                    Stmt (16)
                      Exp (16)
                        ID: yh9YYIYG
                      SEMI
                    StmtList (17)
                      Stmt (17)
                        Exp (17)
                          Exp (17)
                            INT: 3727
                          LB
                          Exp (17)
                            ID: U4poC7pfI
                            LP
                            RP
                          RB
                        SEMI
                      StmtList (18)
                        Stmt (18)
                          RETURN
                          Exp (18)
                            ID: QJAWa
                          SEMI
                RC
            StmtList (20)
              Stmt (20)
                Exp (20)
                  Exp (20)
                    NOT
                    Exp (20)
                      MINUS
                      Exp (20)
                        MINUS
                        Exp (20)
                          ID: S2W13d2UQ0N
                  ASSIGNOP
                  Exp (20)
                    Exp (20)
                      Exp (20)
                        FLOAT: 16.480000
                      AND
                      Exp (20)
                        ID: GZnSUh
                    AND
                    Exp (20)
                      Exp (20)
                        FLOAT: 6556000.000000
                      LB
                      Exp (20)
                        ID: BdFXXIX
                      RB
                SEMI
        RC
    ExtDefList (22)
      ExtDef (22)
        Specifier (22)
          TYPE: int
        FunDec (22)
          ID: Lr4OhDgMQJ
          LP
          VarList (22)
            ParamDec (22)
              Specifier (22)
                StructSpecifier (22)
                  STRUCT
                  OptTag (22)
                    ID: UCB1uF7
                  LC
                  RC
              VarDec (24)
                ID: Hd0RoYT
            COMMA
            VarList (24)
              ParamDec (24)
                Specifier (24)
                  StructSpecifier (24)
                    STRUCT
                    LC
                    RC
                VarDec (26)
                  ID: Eww7fpa
          RP
        CompSt (26)
          LC
          StmtList (27)
            Stmt (27)
              Exp (27)
                Exp (27)
                  Exp (27)
                    ID: Pnpnaru
                  RELOP
                  Exp (27)
                    Exp (27)
                      FLOAT: 68.164001
                    MINUS
                    Exp (27)
                      LP
                      Exp (27)
                        ID: Oru0yKq
                      RP
                OR
                Exp (27)
                  Exp (27)
                    INT: 68096
                  AND
                  Exp (27)
                    FLOAT: 0.060770
              SEMI
            StmtList (28)
              Stmt (28)
                Exp (28)
                  INT: 277
                SEMI
              StmtList (29)
                Stmt (29)
                  Exp (29)
                    Exp (29)
                      Exp (29)
                        MINUS
                        Exp (29)
                          Exp (29)
                            INT: 59
                          LB
                          Exp (29)
                            INT: 952
                          RB
                      OR
                      Exp (29)
                        Exp (29)
                          Exp (29)
                            Exp (29)
                              Exp (29)
                                ID: Fe7EeDV
                                LP
                                RP
                              PLUS
                              Exp (29)
                                Exp (29)
                                  LP
                                  Exp (29)
                                    FLOAT: 330697.000000
                                  RP
                                STAR
                                Exp (29)
                                  FLOAT: 89.199997
                            MINUS
                            Exp (29)
                              ID: JwbWpum
                          RELOP
                          Exp (29)
                            Exp (29)
                              INT: 1312
                            MINUS
                            Exp (29)
                              MINUS
                              Exp (29)
                                NOT
                                Exp (29)
                                  ID: ZMJYsgg
                                  LP
                                  RP
                        RELOP
                        Exp (29)
                          ID: jt9Kb8
                    OR
                    Exp (29)
                      FLOAT: 7.711000
                  SEMI
          RC
      ExtDefList (31)
        ExtDef (31)
          Specifier (31)
            TYPE: float
          SEMI
